### PR TITLE
services/ftp: restricted the anonymous user login

### PR DESCRIPTION
**Pull request by sammynx**

### CODE REVIEW

**Comment by CapacitorSet** [comment]
I think this behaviour should be made optional, as it could allow to collect more data (in the event of a bot that only works with an anonymous login).

File: services/ftp/cmd.go [hunk 1]
```diff
@@ -970,13 +1019,19 @@ func (cmd commandStor) Execute(conn *Conn, param string) {
...
+		return
+	}
+
+	nbytes, err := conn.driver.PutFile(param, conn.dataConn, conn.appendData)
+	if err != nil {
 		conn.writeMessage(450, fmt.Sprintln("error during transfer:", err))
+		return
 	}
+
+	msg := "OK, received " + strconv.Itoa(int(nbytes)) + " bytes"
```
Comment (nl5887) [inline services/ftp/cmd.go]: Could you use `fmt.Sprintf("OK, received %d bytes", " strconv.Itoa(int(nbytes)))` here?

File: services/ftp/cmd.go [hunk 2]
```diff
@@ -671,6 +696,12 @@ func (cmd commandRest) RequireAuth() bool {
 }
 
 func (cmd commandRest) Execute(conn *Conn, param string) {
+
+	if conn.user == "anonymous" {
```
Comment (nl5887) [inline services/ftp/cmd.go]: can we use `IsAnonymous(conn.User)` here? So we can have "anonymous" hardcoded at a single place?

**Comment by nl5887** [comment]
Making the Anonymous account restricted optionally sounds fair. @sammynx could you add an option to that? restrict_anomous?

**Comment by sammynx** [comment]
Created a new PR #213 to address this.
Now you can configure an anonymous user. There is no special treatment for anonymous though.

File: services/ftp/cmd.go (hunk 1 above)
Comment (sammynx) [reply services/ftp/cmd.go]: also fixed in PR213

**Comment by sammynx** [comment]
Close because no longer relevant, because #213

**Pull request closed without merging**